### PR TITLE
UIREQ-931 create JEST/RTL test cases for ComponentToPrint.js

### PR DESCRIPTION
**Pull request by ssandupatla**
UIREQ-931 create JEST/RTL test cases for ComponentToPrint.js
URL: https://issues.folio.org/browse/UIREQ-931

### CODE REVIEW

File: src/components/ComponentToPrint/ComponentToPrint.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const dataSourceMock = {};
+
+  beforeEach(() => {
+    templateFnMock.mockClear();
+  });
+
+  it('renders without crashing', () => {
+    render(<ComponentToPrint templateFn={templateFnMock} dataSource={dataSourceMock} />);
+  });
+
```
Comment (artem-blazhko) [inline src/components/ComponentToPrint/ComponentToPrint.test.js:21]: I am not sure that it is the correct approach.
Test should contain `expect` statement.
Could you remove it?

File: src/components/ComponentToPrint/ComponentToPrint.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  it('renders a Barcode component when encountering a "barcode" tag in the template', () => {
+    const templateFn = jest.fn(() => '<barcode>123456</barcode>');
+    const dataSource = {};
+
+    const { getByTestId } = render(<ComponentToPrint templateFn={templateFn} dataSource={dataSource} />);
+
+    const barcodeComponent = getByTestId('barcode-component');
```
Comment (artem-blazhko) [inline src/components/ComponentToPrint/ComponentToPrint.test.js:28]: Could you follow one approach with constants declaration?
In some places you do not have spaces between constants, in other places you have space.
```suggestion
    const templateFn = jest.fn(() => '<barcode>123456</barcode>');
    const dataSource = {};
    const { getByTestId } = render(<ComponentToPrint templateFn={templateFn} dataSource={dataSource} />);
    const barcodeComponent = getByTestId('barcode-component');
```

**Review by artem-blazhko — CHANGES_REQUESTED** [review]
Could you add changes to the changelog file?
Could you separately test methods inside `rules` array?
Could test scenario when `Component` is null (23 line in the ComponentToPrint.js file)?

**Review by Dmitriy-Litvinenko — CHANGES_REQUESTED** [review]
Could you please update CHANGELOG.md?

**Review by Dmitriy-Litvinenko — COMMENTED** [review]
Could you please fix feedbacks?

**Comment by Dmitriy-Litvinenko** [comment]
Could you please fix feedbacks?

File: src/components/ComponentToPrint/ComponentToPrint.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const parser = new Parser();
+  beforeEach(() => {
+    templateFnMock.mockClear();
+  });
+  it('should handle null result from parseWithInstructions', () => {
+    const originalParseWithInstructions = parser.parseWithInstructions;
+    parser.parseWithInstructions = jest.fn(() => null);
+    render(<ComponentToPrint dataSource={dataSource} templateFn={templateFn} />);
+    expect(screen.queryByTestId('barcode')).not.toBeInTheDocument();
+    parser.parseWithInstructions = originalParseWithInstructions;
```
Comment (artem-blazhko) [inline src/components/ComponentToPrint/ComponentToPrint.test.js:23]: Could you add it to the `afterEach` statement?

File: src/components/ComponentToPrint/ComponentToPrint.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    expect(barcodeElement).toBeInTheDocument();
+  });
+  it('should process the barcode rule with data', () => {
+    const originalParseWithInstructions = parser.parseWithInstructions;
+    const mockParsedComponent = <div data-testid="mock-component">Mock Component</div>;
+    parser.parseWithInstructions = jest.fn(() => mockParsedComponent);
+    render(<ComponentToPrint dataSource={dataSource} templateFn={templateFn} />);
+    const emptyBarcodeElement = screen.getAllByTestId('barcode-component')[1];
+    expect(emptyBarcodeElement).toBeInTheDocument();
+    parser.parseWithInstructions = originalParseWithInstructions;
```
Comment (artem-blazhko) [inline src/components/ComponentToPrint/ComponentToPrint.test.js:46]: the same here

**Review by artem-blazhko — CHANGES_REQUESTED** [review]
Could you add changes to the changelog file?

**Comment by Dmitriy-Litvinenko** [comment]
Could you please fix feedbacks?

**Review by cchitneedi — APPROVED** [review]

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Review by artem-blazhko — APPROVED** [review]

**Pull request merged**